src/semantic_world/world.py

```diff
@@ -1322,6 +1369,54 @@ def compute_split_chain_of_connections(
             tip_connections.append(self.get_connection(tip_chain[i], tip_chain[i + 1]))
         return root_connections, tip_connections
 
+    @property
+    def kinematic_structure_entities_topologically_sorted(
+        self,
+    ) -> List[KinematicStructureEntity]:
+        indices = rx.topological_sort(self.kinematic_structure)
+
+        return [self.kinematic_structure[index] for index in indices]
+
+    @property
+    def bodies_topologically_sorted(self) -> List[Body]:
+        return [
+            entity
+            for entity in self.kinematic_structure_entities_topologically_sorted
+            if isinstance(entity, Body)
+        ]
+
+    def move_subgraph_from_root_to_new_world(
+        self, root: KinematicStructureEntity
+    ) -> World:
+        """
+        Copies the subgraph of the kinematic structure from the root body to a new world and removes it from the old world.
+
+        :param root: The root body of the subgraph to be copied.
+        :return: A new `World` instance containing the copied subgraph.
+        """
+        with self.modify_world():
+            new_world = World(name=self.name)
+            child_bodies = self.compute_descendent_child_kinematic_structure_entities(
+                root
+            )
+            child_body_parent_connections = [
+                body.parent_connection for body in child_bodies
+            ]
+
+            with new_world.modify_world():
```

Comment:
Don't forget the dofs

Reply:
No all dofs are related to connections but some connections can share dofs. But since you are moving the connection and not re-creating it that should be no problem